PocketKit/Sources/PocketKit/Refresh/HomeRefreshCoordinator.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import UIKit
+import Combine
+import Sync
+
+protocol HomeRefreshCoordinatorProtocol {
+    func refresh(isForced: Bool, _ completion: @escaping () -> Void)
+}
+
+class HomeRefreshCoordinator: HomeRefreshCoordinatorProtocol {
+    static let dateLastRefreshKey = "HomeRefreshCoordinator.dateLastRefreshKey"
+    private let notificationCenter: NotificationCenter
+    private let userDefaults: UserDefaults
+    private let source: Source
+    private let refreshRuleTime: Int
+    private var subscriptions: [AnyCancellable] = []
+    private var isRefreshing: Bool = false
+
+    init(notificationCenter: NotificationCenter, userDefaults: UserDefaults, source: Source, refreshRuleTime: Int = 12) {
```

Comment:
As an alternative to `refreshRuleTime` being an `Int` representing hours, we can pass in a minimum refresh interval (of type `TimeInterval`) and use `Date.timeIntervalSince(_:)` for comparison, rather than using calendar components. This would just open us up to more options if we ever change this time in the future. Non-blocker, but a change that might be nice.

Reply:
updated! I like making it more flexible - thanks for the alternative 💯